controllers/jobs.js

```diff
@@ -1,19 +1,70 @@
+const Job = require("../models/Job");
+const { BadRequestError, NotFoundError } = require("../errors");
+const { StatusCodes } = require("http-status-codes");
+
 const getAllJobs = async (req, res) => {
-  res.send("get all jobs");
+  const jobs = await Job.find({ createdBy: req.user.userId }).sort("createdAt");
+  res.status(StatusCodes.OK).json({ jobs, count: jobs.length });
 };
 
 const getJob = async (req, res) => {
-  res.send("get job");
+  console.log(req.user, req.params);
+  const {
+    user: { userId },
+    params: { id: jobId },
+  } = req;
+  console.log("printing data", jobId, userId);
+  const job = await Job.findOne({ _id: jobId, createdBy: userId });
+
+  if (!job) {
+    throw new NotFoundError(`No job with id :  ${jobId}`);
+  }
+  res.status(StatusCodes.OK).json({ job });
 };
 
 const createJob = async (req, res) => {
-  res.json(req.user);
+  req.body.createdBy = req.user.userId;
+  const job = await Job.create(req.body);
+  res.status(StatusCodes.CREATED).json({ job });
 };
+
 const updateJob = async (req, res) => {
-  res.send("update job");
+   console.log(req.user, req.params);
+   const {
+     user: { userId },
+     params: { id: jobId },
+     body: { company, position },
+   } = req;
+   if(company === ""|| position === ""){
```

Comment:
Nitpick:
Since empty double quotes equate to a false value, you can simplify your `if` statement:
```diff
- if (company === ""|| position === "" ){
+ if (!company || !position) {
```